services/app-api/utils/formTemplates/formTemplates.ts

```diff
@@ -31,42 +30,28 @@ export async function getNewestTemplateVersion(reportType: ReportType) {
     Limit: 1,
     ScanIndexForward: false, // true = ascending, false = descending
   };
-  const result = await dynamodbLib.query(queryParams);
-  return result.Items?.[0];
+  return dynamodbLib.query(queryParams);
 }
 
-export const formTemplateForReportType = (reportType: ReportType) => {
-  switch (reportType) {
-    case ReportType.MCPAR:
-      return mcparForm as ReportJson;
-    case ReportType.MLR:
-      return mlrForm as ReportJson;
-    case ReportType.NAAAR:
-      throw new Error(
-        "Not Implemented: NAAAR form template JSON must be added to FormTemplateProvider"
-      );
-    default:
-      assertExhaustive(reportType);
-      throw new Error(
-        "Not Implemented: ReportType not recognized by FormTemplateProvider"
-      );
-  }
-};
-
 export async function getOrCreateFormTemplate(
   reportBucket: string,
   reportType: ReportType
 ) {
-  const currentFormTemplate = formTemplateForReportType(reportType);
-  const formTemplateWithAdminDisabled =
-    copyAdminDisabledStatusToForms(currentFormTemplate);
+  const currentFormTemplate = reportType === "MCPAR" ? mcparForm : mlrForm;
```

Comment:
should we update this to not be a ternary? future-proof it?

Reply:
yeah. it's definitely gonna have to change soon anyway 'cause there's a variant of MCPAR coming soon called PCCM. it won't be a tremendous amount of work but it's gonna be different. i like using a map pattern for this kinda stuff:

```
...
const reportFormMap = {
  MCPAR: mcparForm,
  MLR: mlrForm,
}
const currentFormTemplate = reportFormMap[reportType];
if (!reportFormMap) throw error 
...
```

Reply:
If you look at the removed code on the left half of this file's diff, you'll see `formTemplateForReportType()`, which is precisely the type of future-proofing you're looking for.

I like it better than the map approach because if `if(!reportFormMap)` check is implicit in the switch-case. Plus, it will be enforced by Typescript instantly whenever a new `ReportType` is added.

Reply:
Ooh yeah I like that approach. Let's do it. 👍